rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -503,33 +503,54 @@ protected void registerPegIn(
+                executePegIn(btcTx, peginInformation, totalAmount);
             }
+            case CAN_BE_REFUNDED -> {
+                // If the peg-in cannot be registered means it should be rejected
+                handleRejectedPegin(btcTx,
+                    peginEvaluationResult);
+                logger.debug("[{}] Refunding to address {} ", METHOD_NAME,
+                    peginInformation.getBtcRefundAddress());
+                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTxHash,
+                    totalAmount);
+                markTxAsProcessed(btcTx);
+            }
+            case CANNOT_BE_REFUNDED -> {
+                // If the peg-in cannot be registered means it should be rejected
+                RejectedPeginReason rejectedPeginReason = handleRejectedPegin(btcTx,
+                    peginEvaluationResult);
+
+                handleNonRefundablePegin(btcTx, peginInformation.getProtocolVersion(),
+                    rejectedPeginReason);
 
-            RejectedPeginReason rejectedPeginReason = rejectedPeginReasonOptional.get();
-            logger.debug("[{}] Rejected peg-in, reason {}", METHOD_NAME, rejectedPeginReason);
-            eventLogger.logRejectedPegin(btcTx, rejectedPeginReason);
-            if (peginProcessAction == PeginProcessAction.CAN_BE_REFUNDED) {
-                logger.debug("[{}] Refunding to address {} ", METHOD_NAME, peginInformation.getBtcRefundAddress());
-                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTxHash, totalAmount);
+                if (!activations.isActive(RSKIP459)) {
+                    return;
+                }
+                // Since RSKIP459, rejected peg-ins should be marked as processed
                 markTxAsProcessed(btcTx);
-            } else {
-                logger.debug("[{}] Unprocessable transaction {}.", METHOD_NAME, btcTx.getHash());
-                handleUnprocessableBtcTx(btcTx, peginInformation.getProtocolVersion(), rejectedPeginReason);
             }
         }
     }
 
-    private void handleUnprocessableBtcTx(
+    private RejectedPeginReason handleRejectedPegin(BtcTransaction btcTx,
+        PeginEvaluationResult peginEvaluationResult) {
+        return peginEvaluationResult.getRejectedPeginReason().map(reason -> {
+            logger.debug("[{handleRejectedPegin}] Rejected peg-in, reason {}", reason);
+            eventLogger.logRejectedPegin(btcTx, reason);
+            return reason;
+        }).orElseThrow(() -> {
+            // This flow should never be reached. There should always be a rejected pegin reason.
+            String message = "Invalid state. No rejected reason was returned for an invalid pegin.";
+            logger.error("[{handleRejectedPegin}] {}", message);
+            return new IllegalStateException(message);
+        });
+    }
+
+    private void handleNonRefundablePegin(
```

Comment:
Perhaps the same renaming idea could apply here

Reply:
Done